interp/notation_ops.ml

```diff
@@ -1074,6 +1074,7 @@ let rec match_ inner u alp metas sigma a1 a2 =
       let alp,sigma = Array.fold_right2 (fun id1 id2 alsig ->
 	match_names metas alsig (Name id1) (Name id2)) idl1 idl2 (alp,sigma) in
       Array.fold_left2 (match_in u alp metas) sigma bl1 bl2
+  (* Why this special case ? *)
   | GCast(_,c1,CastConv t1), NCast (c2,CastConv t2)
   | GCast(_,c1,CastVM t1), NCast (c2,CastVM t2) ->
       match_in u alp metas (match_in u alp metas sigma c1 c2) t1 t2
```

Comment:
Would you like to have a sub-function `match_cast`? That would be fine to me. At least, you're right that the `CastNative` case is missing.

BTW, introducing a `match_cast` function would have the additional benefit that when a new `cast_type` is added, it is detected, while the catch-remaining cases clause that you added below would not detect a new `cast_type`.